src/Features/Core/Portable/ExternalAccess/UnitTesting/UnitTestingFeaturesReferencesService.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting;
+
+internal interface IUnitTestingCodeLensContext
+{
+    Task<ImmutableArray<ReferenceMethodDescriptor>?> FindReferenceMethodsAsync(
+        Guid projectGuid, string filePath, TextSpan span, DocumentId? sourceGeneratedDocumentId, CancellationToken cancellationToken);
+}
+
+internal interface IUnitTestingFeaturesReferencesServiceCallback
+{
+    Task<TResult> InvokeAsync<TResult>(string targetName, IReadOnlyList<object?> arguments, CancellationToken cancellationToken);
+}
```

Comment:
you'll provide an impl of this on your side.  it will be implemented as:

```C#
class YourImpl(
    IAsyncCodeLensDataPointProvider provider,
    ICodeLensCallbackService callbackService) : IUnitTestingFeaturesReferencesServiceCallback
{
    public Task<TResult> InvokeAsync<TResult>(string targetName, IReadOnlyList<object?> arguments, CancellationToken cancellationToken) => callbackService.InvokeAsync(targetName, arguments, cancellationToken);
}
```